docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,280 @@
+                   Successive calls of this request keep the microVM in the `Paused`
+                   state.
+**Effects**:
+- _on success_: microVM is guaranteed to be `Paused`.
+- _on failure_: no side-effects.
+
+*Note*: If you change your mind and want to create a snapshot later, you can resume the
+        microVM and then pause it again when you decide to take that snapshot.
+
+## Creating snapshots
+
+Now that the microVM is paused, you can create a snapshot, which can be either a `full`
+one or a `diff` one. Full snapshots always create a complete, resume-able snapshot of
+the current microVM state and memory. Diff snapshots save the current microVM state
+and the memory dirtied since the last snapshot (full or diff). Diff snapshots are not
+resume-able, but can be merged into a full snapshot using external (provided) tooling.
+
+### Creating full snapshots
+
+For creating a full snapshot, you can use the following API command:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PUT "http://localhost/snapshot/create" \
+    -H  "accept: application/json" \
+    -H  "Content-Type: application/json" \
+    -d "{
+            \"snapshot_type\": \"Full\",
+            \"snapshot_path\": \"/path/to/snapshot_file\",
+	        \"mem_file_path\": \"/path/to/mem_file\",
+            \"version\": \"2\"
+    }"
+```
+
+Details about the required and optional fields can be found in the
+[swagger definition](../../src/api_server/swagger/firecracker.yaml).
+
+**Prerequisites**: It can be called only when the microVM is `Paused`.
+**Effects**:
+- _on success_:
+  - The file indicated by `/path/to/snapshot_file` contains the devices' model state
+    and emulation state. The one indicated by `/path/to/mem_file` contains a full copy
+    of the guest memory.
+  - The block devices contents are flushed to their backing files. At this point, these
+    should be backed up externally by the user. The generated snapshot files are
+    immediately available to be used (current process releases ownership).
+  - The snapshot creation resets then the dirtied page bitmap and marks all pages clean
+    (from a diff snapshot point of view).
+
+If a `version` is specified, the new snapshot is saved at that version, otherwise
+it will be saved at the same version of the running Firecracker. The version is only
+used for the microVM state file as it contains internal state structures for device
+emulation, vCPUs and others that can change their format from a Firecracker version
+to another. Versioning is not supported for block devices and memory files. The separate
+block device file components of the snapshot have to be handled by the user.
+
+- _on failure_: no side-effects.
+
+### Creating diff snapshots
+
+For creating a diff snapshot, you should use the same API command, but with
+`snapshot_type` field set to `Diff`.
+*Note*: If omitted from the request body, `snapshot_type` default value is `Full`.
```

Comment:
```suggestion
*Note*: If not specified, `snapshot_type` defaults to`Full`.
```

Reply:
`defaults` in this context is a phrasal verb and I don't particularly like using it in a doc; replaced it with `is by default`.